io/fs/ffsb.py

```diff
@@ -0,0 +1,232 @@
+                    dirnm = ''.join(random.choice(string.letters)
+                                    for i in xrange(9))
+                    if line[m.end(2) - 1] == '/':
+                        newline = '%s%s' % (line[0:m.end(2)], dirnm)
+                        ffsbdir = '%s%s' % (line[m.start(2):m.end(2)], dirnm)
+                    else:
+                        newline = '%s/%s' % (line[0:m.end(2)], dirnm)
+                        ffsbdir = '%s/%s' % (line[m.start(2):m.end(2)], dirnm)
+                    self.tempdirs.append(ffsbdir)
+                    if os.path.exists(ffsbdir):
+                        continue
+                    else:
+                        os.makedirs(ffsbdir)
+                        break
+                fw.write(newline + '\n')
+                continue
+            p = re.compile(r'(\s*\t*num_files)\=(.*)')
+            m = p.match(line)
+            if m:
+                newline = '%s=%s' % (line[0:m.end(1)], str(params[fsno][2]))
+                fw.write(newline + '\n')
+                continue
+            p = re.compile(r'(\s*\t*max_filesize)\=(\d+[kKMmGgTt]?)')
+            m = p.match(line)
+            if m:
+                newline = '%s%s' % (line[0:m.start(2)], str(params[fsno][3]))
+                fw.write(newline + '\n')
+                continue
+            fw.write(line + '\n')
+        fr.close()
+        fw.close()
+
+    def setUp(self):
+        """
+        Build 'ffsb'.
+        """
+
+        self.update_config('profile.cfg.sample')
+
+        tarball = self.fetch_asset('http://prdownloads.sourceforge.net'
+                                   '/ffsb/ffsb-6.0-rc2.tar.bz2')
+        data_dir = os.path.abspath(self.datadir)
+        archive.extract(tarball, self.srcdir)
+        ffsb_version = os.path.basename(tarball.split('.tar.')[0])
+        self.srcdir = os.path.join(self.srcdir, ffsb_version)
+
+        patch = self.params.get('patch',
+                                default='ffsb-ppc64le-arch-support.patch')
+
+        os.chdir(self.srcdir)
+        p1 = 'patch -p1 < %s/%s' % (data_dir, patch)
+        process.run(p1, shell=True)
+        process.run('[ -x configure ] && ./configure', shell=True)
+        build.make(self.srcdir)
+
+    def update_config(self, cfg):
+        """
+        Update the profile.cfg file.
+
+        :param cfg: Basename of the cfg file, that should be on the
+                    test module folder (client/tests/ffsb) or URL of the
+                    remote config file.
+        """
+
+        filename = os.path.basename(cfg)
+        profile_src = os.path.join(self.datadir, filename)
+        profile_dst = os.path.join(os.path.dirname(self.datadir),
+                                   'profile.cfg')
+        shutil.copyfile(profile_src, profile_dst)
+
+    def test(self, cfg=None):
+        """
+        Runs a single iteration of the FFSB.
+        """
+        profile_cfg = self.params.get('profile_cfg', default='profile.cfg')
```

Comment:
As long as we handle the profile.cfg file properly, we don't need to change it here. So we can always use `profile.cfg` to run ffsb, since `profile.cfg` will be a copy out of `profile.cfg.default` OR from the user custom profile.cfg file (provided through yaml file).